tests/test_parser.py

```diff
@@ -0,0 +1,164 @@
+        """This is the summary.
+
+        This is the description.
+
+        Args:
+            x: The first argument.
+            y: The second argument.
+
+        Returns:
+            The sum of x and y.
+        """
+        return x + y
+
+    definition = _parse_func(f)
+
+    assert definition == PyFunc(
+        name="test_parser.test_parse_func.<locals>.f",
+        signature="test_parser.test_parse_func.<locals>.f(x: int, y: int) -> int",
+        summary="This is the summary.",
+        desc="This is the description.",
+        args=[
+            {"name": "x", "type": None, "desc": "The first argument."},
+            {"name": "y", "type": None, "desc": "The second argument."},
+        ],
+        returns="The sum of x and y.",
+        examples=[],
+    )
+
+
+def test_parse_func_multi_line_summary():
+    def f(x: int, y: int) -> int:
+        """
+        Summary SummarySummarySummarySummarySummarySummarySummarySummarySummarySummarySummary
+        SummarySummary
+
+        This is the description.
+
+        Args:
+            x: The first argument.
+            y: The second argument.
+
+        Returns:
+            The sum of x and y.
+        """
+        return x + y
+
+    definition = _parse_func(f)
+
+    assert definition == PyFunc(
+        name="test_parser.test_parse_func_multi_line_summary.<locals>.f",
+        signature="test_parser.test_parse_func_multi_line_summary.<locals>.f(x: int, y: int) -> int",
+        summary="Summary SummarySummarySummarySummarySummarySummarySummarySummarySummarySummarySummary SummarySummary",
+        desc="This is the description.",
+        args=[
+            {"name": "x", "type": None, "desc": "The first argument."},
+            {"name": "y", "type": None, "desc": "The second argument."},
+        ],
+        returns="The sum of x and y.",
+        examples=[],
+    )
+
+
+def test_parse_func_multi_line_summary_first_line():
+    def f(x: int, y: int) -> int:
+        """Summary SummarySummarySummarySummarySummarySummarySummarySummarySummarySummarySummary
+        SummarySummarySummary
+
+        This is the description.
+
+        Args:
+            x: The first argument.
+            y: The second argument.
+
+        Returns:
+            The sum of x and y.
+        """
+        return x + y
+
+    definition = _parse_func(f)
+
+    assert definition == PyFunc(
+        name="test_parser.test_parse_func_multi_line_summary_first_line.<locals>.f",
+        signature="test_parser.test_parse_func_multi_line_summary_first_line.<locals>.f(x: int, y: int) -> int",
+        summary="Summary SummarySummarySummarySummarySummarySummarySummarySummarySummarySummarySummary SummarySummarySummary",
+        desc="This is the description.",
+        args=[
+            {"name": "x", "type": None, "desc": "The first argument."},
+            {"name": "y", "type": None, "desc": "The second argument."},
+        ],
+        returns="The sum of x and y.",
+        examples=[],
```

Comment:
Since the intent for this and the following tests is to verify the summary and description, I think we can simplify these tests by only checking those two attributes.

```suggestion
    assert definition.summary == "Summary SummarySummarySummarySummarySummarySummarySummarySummarySummarySummarySummary SummarySummarySummary"
    assert definition.desc ==" This is the description."
```